psydac/linalg/stencil.py

```diff
@@ -858,6 +895,29 @@ def dot( self, v, out=None):
         out.ghost_regions_in_sync = False
         return out
 
+    def vdot( self, v, out=None):
```

Comment:
I am not sure if this function is necessary. I have implemented it because if we have to do (A_*x) and we want to do it efficiently, unless writing this : 
`(A.dot(x.conjugate())).conjugate()`

We can write : 
`A.vdot(x)`

I thought it would made the code more understandable

Reply:
Yep it's done :
[90f3320](https://github.com/pyccel/psydac/pull/273/commits/90f3320144dbbcbb5eff590319d36d866bc35641)